src/app/projects/[projectId]/components/UnsavedChangesToast.Client.tsx

```diff
@@ -0,0 +1,95 @@
+"use client"
+
+import Image from "next/image"
+import { useEffect, useRef, useState } from "react"
+import { toast, Toaster } from "sonner"
+
+import MetadataPublishedConfirmationDialog from "@/components/projects/publish/MetadataPublishedConfirmationDialog"
+import { createProjectSnapshot } from "@/lib/actions/snapshots"
+import { ProjectWithDetails } from "@/lib/types"
+import { projectHasUnpublishedChanges } from "@/lib/utils"
+import { useAnalytics } from "@/providers/AnalyticsProvider"
+
+const UnsavedChangesToastClient = ({
+  project,
+}: {
+  project: ProjectWithDetails
+}) => {
+  const { track } = useAnalytics()
+  const toastOpenRef = useRef(false)
+  const [showMetadataPublishedDialogue, setShowMetadataPublishedDialogue] =
+    useState(false)
+  const hasUnpublishedChanges = projectHasUnpublishedChanges(project)
+  const hasBeenPublished = project ? project?.snapshots.length > 0 : false
+
+  const onPublish = async () => {
+    toast.promise(createProjectSnapshot(project.id), {
+      loading: "Publishing metadata onchain...",
+      success: ({ snapshot }) => {
+        setShowMetadataPublishedDialogue(true)
+        track("Publish Project", {
+          projectId: project.id,
+          attestationId: snapshot?.attestationId,
+        })
+        return "Snapshot published"
+      },
+      error: () => {
```

Comment:
it might be that this error callback is incorrectly inferring the non-error response

Reply:
hm this is actually very odd -- the behavior from this toast library specifies that it will throw this error block if the promise rejects. I looked deeper into the `createProjectSnapshot` function, and it never rejects or throws. Instead, it we hit an error, we are returning something like `{error: "message"}` which still would resolve the promise as a success. Anyways, I'm confused how it's ever hitting this block now .... 🤔

Reply:
yea weirdly, locally it's working fine. But not in the preview...